### DISCUSSION TITLE
How can I actually get thread.archiver_id ?

### DISCUSSION THREAD

**Discussion by Gooraeng** [Rapptz/discord.py#9850] [Q&A] [answered]
How are you? I always appreciate to all contributors for creating and making improved this amazing library.
I am beginner developer and I want to try to ask QnA about thread.archiver_id.
As far as I know, in discord.py document, it could be wrong, I can retrive archive_id when a thread is closed. But, It always returns None even if I have proper intents. 

And today, I found there is no information about archiver_id in discord api docs [https://discord.com/developers/docs/resources/channel#thread-metadata-object](here). or are there ways to get it?
Thank you for all your advises

**Comment by tailoric** [comment, marked as answer]
Hello, yes it seems like that information is not in the thread metadata
according to this commit:
https://github.com/discord/discord-api-docs/commit/82c98ba5699b69706cfb493a7c6e08b67ae1a8f3
You should use the audit log as the source of truth 
So you probably will need an listener for [audit_log_entry_create](https://discordpy.readthedocs.io/en/latest/api.html?highlight=on_audit_log#discord.on_audit_log_entry_create)
Edit: [Or just iterate through the Guild.audit_logs](https://discordpy.readthedocs.io/en/latest/api.html?highlight=guild%20audit_log#discord.Guild.audit_logs) (however discord doesn't keep entries indefinitely, so you need to store them in some way if you need them for longer)
and then check for [thread updates](https://discordpy.readthedocs.io/en/latest/api.html?highlight=on_audit_log#discord.AuditLogAction.thread_update)

**Reply by Gooraeng** [reply]
thank you so much for paying attention to me. i solved my proplem finally!!
